### PR TITLE
Patch feedstock for cross-compiling on osx + arm architecture

### PR DESCRIPTION
**Pull request by satra**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [X] Reset the build number to `0` (if the version changed)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

Fixes #87

### CODE REVIEW

File: recipe/osx-arm64-hdf5.diff [hunk 1]
```diff
@@ -0,0 +1,12 @@
+--- a/setup_configure.py
++++ b/setup_configure.py
+@@ -68,6 +68,9 @@ class BuildConfig:
+         mpi = mpi_enabled()
+         h5_inc, h5_lib, h5_macros = cls._find_hdf5_compiler_settings(mpi)
+
++        if os.environ.get('HDF5_HAS_ROS3', None) == '1':
++            return cls(h5_inc, h5_lib, h5_macros, (1, 10, 6), mpi, True)
```
Comment (isuruf) [inline recipe/osx-arm64-hdf5.diff]: Please use `HDF5_VERSION` env variable instead of hard coding 1.10.6.

**Comment by satra** [comment]
@jakirkham any ideas about the travis failure. Looks more like a CI issue than h5py

**Comment by satra** [comment]
@isuruf and @jakirkham -  any thoughts on the travis issue. i posted on gitter. don't know what else to do.

**Comment by isuruf** [comment]
Restarted

**Comment by satra** [comment]
@isuruf - looks like this is all green. are there any other changes needed, or can this be merged?

**Comment by jakirkham** [comment]
@djhoese, could you please take a look? Think the ROS3 portion may be relevant to you

**Comment by djhoese** [comment]
I'm not sure I understand the issue, but what I think I understand is that the way conda-forge is compiling things makes it so h5py can't determine the HDF5 version with the way HDF5 is being built. Right? So this patch doesn't turn off/disable ROS3, but works around the compilation issues?

**Comment by isuruf** [comment]
Yes. To be precise, conda-forge is cross compiling for macos-arm64 from macos-x86_64, so h5py can't figure out whether hdf5 was compiled with ROS3 or not. We are just patching h5py to say that hdf5 was compiled with ROS3

**Comment by satra** [comment]
@djhoese - yes, that is correct.

this patch is specifically for the cross-compilation that is happening for osx arm architectures. since this is conda dependent, we are making the assumption that the conda hdf5 libraries already have ROS3 support. so it simply patches the build process to give it the version of the library, without trying to load the library to detect flags, which is what the h5py build normally does.

**Comment by djhoese** [comment]
Sounds good. Thanks everyone. FYI although I added the ROS3 build flags to HDF5, I don't actively use it or know much about it.

**Comment by jakirkham** [comment]
Thanks Satrajit for the PR! Also thanks Isuru and David for the reviews! 😄

**Comment by satra** [comment]
thank you @jakirkham, @isuruf , and @djhoese

**Pull request merged**